fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionDataValidator.java

```diff
@@ -271,6 +291,10 @@ public SavingsAccountTransaction validateHoldAndAssembleForm(final String json,
 
         SavingsAccountTransaction transaction = SavingsAccountTransaction.holdAmount(account, account.office(), paymentDetails,
                 transactionDate, Money.of(account.getCurrency(), amount), createdDate, createdUser);
+
```

Comment:
Generally speaking I don't like that a validator is actually creating new domain objects like a SavingsAccountTransaction. Seems like we have given the validator way too much responsibility and I'd prefer to split it up.

I see it wasn't this PR that introduced this approach so I won't block it because of this but would be really beneficial to divide these.